azext_iot/tests/iothub/message_endpoint/test_iothub_message_endpoint_unit.py

```diff
@@ -0,0 +1,938 @@
+                "resource_group_name": generate_names(),
+            },
+            {
+                "endpoint_account_name": None,
+                "endpoint_resource_group": None,
+                "endpoint_subscription_id": None,
+                "endpoint_policy_name": None,
+                "connection_string": None,
+                "endpoint_uri": generate_names(),
+                "entity_path": generate_names(),
+                "identity": generate_names(),
+                "resource_group_name": None,
+            },
+            {
+                "endpoint_account_name": generate_names(),
+                "endpoint_resource_group": None,
+                "endpoint_subscription_id": None,
+                "endpoint_policy_name": generate_names(),
+                "connection_string": "update",
+                "endpoint_uri": generate_names(),
+                "entity_path": generate_names(),
+                "identity": None,
+                "resource_group_name": generate_names(),
+            },
+            {
+                "endpoint_account_name": None,
+                "endpoint_resource_group": generate_names(),
+                "endpoint_subscription_id": None,
+                "endpoint_policy_name": generate_names(),
+                "connection_string": None,
+                "endpoint_uri": generate_names(),
+                "entity_path": None,
+                "identity": generate_names(),
+                "resource_group_name": None,
+            },
+            {
+                "endpoint_account_name": generate_names(),
+                "endpoint_resource_group": generate_names(),
+                "endpoint_subscription_id": generate_names(),
+                "endpoint_policy_name": generate_names(),
+                "connection_string": generate_names(),
+                "endpoint_uri": generate_names(),
+                "entity_path": generate_names(),
+                "identity": None,
+                "resource_group_name": generate_names(),
+            },
+        ]
+    )
+    def test_message_endpoint_update_event_hub(self, mocker, fixture_cmd, fixture_update_endpoint_ops, req):
+        result = subject.message_endpoint_update_event_hub(
+            cmd=fixture_cmd,
+            hub_name=hub_name,
+            endpoint_name=endpoint_name,
+            **req
+        )
+        fixture_find_resource, fixture_arg_check = fixture_update_endpoint_ops
+
+        assert result == generic_response
+        resource_group = fixture_find_resource.call_args[0][2]
+        assert req.get("resource_group_name") == resource_group
+        hub_resource = fixture_find_resource.call_args[0][0].client.begin_create_or_update.call_args[0][2]
+        endpoints = hub_resource.properties.routing.endpoints.event_hubs
+        assert len(endpoints) == 1
+        endpoint = endpoints[0]
+
+        assert endpoint.name == endpoint_name
+        mock = mocker.Mock
+
+        # if a prop is not set, it will be a Mock object
```

Comment:
The idea behind these checks is that if a property of an endpoint is not modified by update, it will show up as a mock object. Are you all ok with this